Fix updateName function

chriscarrollsmith opened the pull request:
- Fixes #311 and closes #310: Alters `updateName` function so it changes the public.users table rather than the auth.users table (which was problematic for reasons laid out in the issue)
- Fixes #312: Removes `--turbo` flag from `dev` command in package.json, which was breaking some dependencies (UPDATE: we're keeping --turbo but updating to next@canary)
- Adds support for generating SQL queries locally with Supabase AI
- Improves documentation for spinning up a local Supabase instance (because the previous documentation neglected to explain that you have to set variables in `.env` or setup will fail)
- Updates dependencies
- UPDATE: Fixes #297: Restores missing types for two columns in the `prices` table
- UPDATE: Adds mobile-friendly responsive hamburger menu
- UPDATE: Enhances documenation

leerob commented on package.json:
```diff
@@ -3,14 +3,15 @@
   "version": "0.0.0",
   "license": "MIT",
   "scripts": {
-    "dev": "next dev --turbo",
```
Let's keep this and please share the errors you're seeing! We'd like to fix them. Could you also try on the latest `next@canary` please?

chriscarrollsmith replied:
Sure. Here's the error:

![Screenshot 2024-03-11 105127](https://github.com/vercel/nextjs-subscription-payments/assets/75859865/8d581064-2218-466d-ac31-e2be2521e880)

I'm happy to report that it seems fixed in next@canary, though! Should I update the PR to use that, or should I wait until official release of stable 14.2.0?

chriscarrollsmith commented:
PR updated as follows:

- Next.js version bump to next@canary
- Restored `--turbo` flag
- Fixed missing types
- Added mobile-friendly responsive hamburger menu

leerob commented:
Yeah, go ahead and keep `canary` 😄

k-thornton commented on README.md:
```diff
@@ -95,7 +97,7 @@ For example, you can create business models with different pricing tiers, e.g.:
   - Price 2: 200 USD per year
 
-Optionally, to speed up the setup, we have added a [fixtures file](fixtures/stripe-fixtures.json) to bootstrap test product and pricing data in your Stripe account. The [Stripe CLI](https://stripe.com/docs/stripe-cli#install) `fixtures` command executes a series of API requests defined in this JSON file. Simply run `stripe fixtures fixtures/stripe-fixtures.json`.
+Optionally, to speed up the setup, we have added a [fixtures file](fixtures/stripe-fixtures.json) to bootstrap test product and pricing data in your Stripe account. The [Stripe CLI](https://stripe.com/docs/stripe-cli#install) `fixtures` command executes a series of API requests defined in this JSON file. Simply run `stripe fixtures fixtures/stripe-fixtures.json`. (Note the use of a numerical inde ...
```
omg thank you for this note!  You just helped me figure out how to sequence my products!

chriscarrollsmith commented:
You're welcome! :)

On Fri, May 10, 2024, 2:51 PM Kyle Thornton ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In README.md
> <https://github.com/vercel/nextjs-subscription-payments/pull/313#discussion_r1597256607>
> :
>
> > @@ -95,7 +97,7 @@ For example, you can create business models with different pricing tiers, e.g.:
>    - Price 1: 20 USD per month
>    - Price 2: 200 USD per year
>
> -Optionally, to speed up the setup, we have added a [fixtures file](fixtures/stripe-fixtures.json) to bootstrap test product and pricing data in your Stripe account. The [Stripe CLI](https://stripe.com/docs/stripe-cli#install) `fixtures` command executes a series of API requests defined in this JSON file. Simply run `stripe fixtures fixtures/stripe-fixtures.json`.
> +Optionally, to speed up the setup, we have added a [fixtures file](fixtures/stripe-fixtures.json) to bootstrap test product and pricing data in your Stripe account. The [Stripe CLI](https://stripe.com/docs/stripe-cli#install) `fixtures` command executes a series of API requests defined in this JSON file. Simply run `stripe fixtures fixtures/stripe-fixtures.json`. (Note the use of a numerical index in the product metadata to control display sequence in the UI).
>
> omg thank you for this note! You just helped me figure out how to sequence
> my products!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vercel/nextjs-subscription-payments/pull/313#pullrequestreview-2050916913>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASCYPGPDDYO2M7HMW6S2CYLZBU6NNAVCNFSM6AAAAABEOTL22OVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDANJQHEYTMOJRGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>
